### PR TITLE
Improve AsyncPipe

### PR DESCRIPTION
**Pull request by kennylavender**
[PR Process](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-process) - [PR Review Checklist](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-review-checklist)

### Release
 Semantic release is enabled for this repository. Make sure you follow the right commit message convention. 
 We're using semantic-release's default — [Angular Commit Message Conventions](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines).

## Description of Changes

- better types
- fix implementation and types for no initial value
- had to switch to function declaration implementation for overloads to be accepted properly

### CODE REVIEW

File: src/Helpers/async-pipe.test.ts [hunk 1]
```diff
@@ -31,4 +32,13 @@ describe('asyncPipe', async (assert: any) => {
       expected: await incP(double(inc(await doubleP(value)))),
     })
   }
+
+  {
+    assert({
+      given: 'no starting value',
+      should,
+      actual: await asyncPipe(fixedNum, incP)().catch(identity),
```
Comment (lautarodragan) [inline src/Helpers/async-pipe.test.ts]: Is the catch here needed?

Comment (kennylavender) [reply src/Helpers/async-pipe.test.ts]: Not really for this particular test, we do need a test for it though to make sure the types don't complain, I will remove the catch from all the current tests and make a specific test.

File: src/Helpers/async-pipe.ts [hunk 2]
```diff
@@ -1,4 +1,86 @@
-type Func = (v?: any) => any | Promise<any>
+export function asyncPipe<V0 extends any, T1>(fn1: (x?: V0) => T1 | Promise<T1>): (x?: V0) => Promise<T1>
```
Comment (lautarodragan) [inline src/Helpers/async-pipe.ts]: Use `asyncPipe` as a name (following AirBnB's naming standard for files)

Comment (kennylavender) [reply src/Helpers/async-pipe.ts]: will rename in follow up PR 👍

**Pull request merged**